lib/ABI/RegisterStateDeductions.cpp

```diff
@@ -52,18 +51,22 @@ size_t findLastUsedIndex(llvm::ArrayRef<model::Register::Values> Registers,
   return 0;
 }
 
-template<model::ABI::Values ABI, bool EnforceABIConformance>
+template<bool EnforceABIConformance>
 struct DeductionImpl {
-  static std::optional<StateMap> run(const StateMap &InputState) {
-    if constexpr (abi::Trait<ABI>::ArgumentsArePositionBased)
+  const abi::Definition &ABI;
+  const std::string_view ABIName;
+  explicit DeductionImpl(const abi::Definition &ABI) :
+    ABI(ABI), ABIName(model::ABI::getName(ABI.ABI())) {}
+
+  std::optional<StateMap> run(const StateMap &InputState) {
+    if (ABI.ArgumentsArePositionBased())
```

Comment:
Uhm AFAIU this change is required to let previous commits compile.
Do all commits build correctly?
If not, we need a bit more squashing.

Same for:
* ABI: move RegisterOrder onto the runtime trait 
* ABI: move function prototype onto the runtime trait 
* ABI: move ToRawConverter onto the runtime trait 
* ABI: move ToCABIConverter onto the runtime trait 

Aaah, I see, for a while you keep both.
I would have been fine with a single commit for all of this, but since it's there, it can stay (as long as all commits still build).

Reply:
I ran `orc check-branch`. It's all good.

Yeah, I use the trick of not removing old stuff until it's fully reworked quite often.

P. S. It did backfire somewhat in this case though. There are some functions in `Support.h` that shouldn't be there in the and.